dowhy/gcm/density_estimator.py

```diff
@@ -0,0 +1,15 @@
+from abc import ABC, abstractmethod
+
+import numpy as np
+
+
+class DensityEstimator(ABC):
+
+    @abstractmethod
+    def fit(self, X: np.ndarray) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    def estimate_density(self, X: np.ndarray) -> np.ndarray:
```

Comment:
to be consistent with our naming convention, shall we simply name this method "density"?

Reply:
Renamed to ```density```.